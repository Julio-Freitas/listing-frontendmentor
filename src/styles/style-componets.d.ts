import { Theme } from './theme';

type Theme = typeof Theme;
declare module 'styled-components' {
  export interface DefaultTheme extends Theme {}
}
