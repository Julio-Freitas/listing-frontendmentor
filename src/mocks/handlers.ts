import { rest } from 'msw';
import { listingData, listingEmptyData } from './responses/listing.data';

export const handlers = [
  rest.get('http://localhost:3000/api/list', (req, res, ctx) =>
    res(ctx.status(200), ctx.json(listingData))
  )
];

export const listDataHandlers = [
  rest.get('http://localhost:3000/api/list', (req, res, ctx) =>
    res(ctx.status(200), ctx.json(listingData))
  )
];
export const listDataFailHandlers = [
  rest.get('http://localhost:3000/api/list', (req, res, ctx) =>
    res(ctx.status(400), ctx.json(listingEmptyData))
  )
];
