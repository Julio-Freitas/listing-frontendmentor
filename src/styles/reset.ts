import { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
    // this is the shared style
    html, body {
    box-sizing: border-box;
    font-size: 62.5%;
    font-family: ${({ theme }) => theme.fonts.primary.leagueSpartan};
    background-color:${({ theme }) =>
      theme.colors.secondary.lightGrayishCyanBackground};
    margin: 0 auto;

    }

    #__next {
      display: flex;
      align-items: center;
      justify-content: center;
      flex-direction: column;
      max-width: 1140px;
      margin: 0 auto;
    }

    #__next main {
        display: flex;
        flex-direction: column;
        height: 100%;
        width: 100%;
    }
    * {
        box-sizing: border-box;
        margin: 0;
        padding: 0;
    }
    a{
        text-decoration: none;
    }

    input {
        all: unset;
    }

    button {
        cursor: pointer
    }
    img{
      display: inline-block;
      max-width:100%;
      margin:0;
    }
`;
