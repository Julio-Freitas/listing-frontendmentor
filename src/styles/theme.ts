export const Theme = {
  fonts: {
    primary: {
      leagueSpartan: 'League Spartan, sans-serif'
    },
    secondary: {
      lexendDeca: 'Lexend Deca, sans-serif'
    }
  },
  colors: {
    primary: {
      darkCyan: 'hsl(180, 29%, 50%)'
    },
    secondary: {
      lightGrayishCyanBackground: 'hsl(180, 52%, 96%)',
      lightGrayishCyanFilter: 'hsl(180, 31%, 95%)',
      darkGrayishCyanFilter: 'hsl(180, 8%, 52%)',
      veryDarkGrayinshCyan: 'hsl(180, 14%, 20%)'
    }
  }
};
