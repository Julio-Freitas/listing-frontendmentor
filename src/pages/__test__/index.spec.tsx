import React, { ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';
import { act, render, screen, waitFor } from '@testing-library/react';
import { listDataFailHandlers } from '../../mocks/handlers';
import { server } from '../../mocks/server';
import { Theme } from '../../styles/theme';
import Home from '..';
import userEvent from '@testing-library/user-event';

const renderWithProviderTheme = (component: ReactNode) => {
  return {
    ...render(<ThemeProvider theme={Theme}>{component}</ThemeProvider>)
  };
};

describe('Testing home', () => {
  it('should render conrrectly', async () => {
    const { container } = await act(async () => {
      return renderWithProviderTheme(<Home />);
    });

    expect(screen.queryByTestId('header')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render when click in any tags', async () => {
    await act(async () => {
      return renderWithProviderTheme(<Home />);
    });
    const list = screen.getByTestId('skills');
    const header = screen.getByTestId('header');

    userEvent.click(list.children[0]);
    expect(header.childElementCount).toBe(1);

    await waitFor(() => {
      const tagHeader = screen.getAllByRole('listitem-skills');
      expect(tagHeader[0]).toHaveTextContent(/html/gi);
    });

    expect(header.childElementCount).toBe(2);
    expect(list).toHaveTextContent(/Html/gi);
    expect(screen.queryByTestId('header')).toBeInTheDocument();
  });

  it('should render when click in tags of header', async () => {
    await act(async () => {
      return renderWithProviderTheme(<Home />);
    });
    const list = screen.getByTestId('skills');
    const header = screen.getByTestId('header');

    userEvent.click(list.children[0]);

    await waitFor(async () => {
      const tagHeader = screen.getAllByRole('listitem-skills');
      expect(tagHeader[0]).toHaveTextContent(/html/gi);
      expect(header.childElementCount).toBe(2);
      await userEvent.click(tagHeader[0]);
    });

    expect(header.childElementCount).toBe(1);
    expect(list).toHaveTextContent(/Html/gi);
    expect(screen.queryByTestId('header')).toBeInTheDocument();
  });

  it('should render conrrectly when click', async () => {
    await act(async () => {
      server.use(...listDataFailHandlers);
      renderWithProviderTheme(<Home />);
    });
    expect(screen.getByTestId('header')).toBeInTheDocument();
    expect(screen.getByText('Algo de errado aconteceu!')).toBeInTheDocument();
  });
});
