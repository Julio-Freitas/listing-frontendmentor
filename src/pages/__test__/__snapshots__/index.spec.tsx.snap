// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing home should render conrrectly 1`] = `
.c0 {
  background: hsl(180,29%,50%);
  width: 100%;
  height: 124px;
  position: relative;
  margin-bottom: 2.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 div:first-child {
  background-image: url('/images/bg-header-desktop.svg');
  width: 100%;
  height: 100%;
  background-repeat: no-repeat;
  background-size: cover;
}

<header
  class="c0"
  data-testid="header"
>
  <div />
</header>
`;
